src/daft-core/src/array/ops/is_in.rs

```diff
@@ -0,0 +1,92 @@
+use crate::{
+    array::DataArray,
+    datatypes::{
+        BinaryArray, BooleanArray, DaftIntegerType, DaftNumericType, Float32Array, Float64Array,
+        NullArray, Utf8Array,
+    },
+    DataType,
+};
+
+use super::{full::FullNull, DaftIsIn};
+use common_error::DaftResult;
+use std::hash::Hash;
+
+use super::as_arrow::AsArrow;
+
+macro_rules! collect_to_hashset_and_check_membership {
+    ($self:expr, $rhs:expr) => {{
+        let hashset = $rhs
+            .as_arrow()
+            .iter()
+            .collect::<std::collections::HashSet<_>>();
+        let result = $self
+            .as_arrow()
+            .iter()
+            .map(|x| hashset.contains(&x))
+            .collect::<Vec<_>>();
+        Ok(BooleanArray::from(($self.name(), result.as_slice())))
+    }};
+}
+
+impl<T> DaftIsIn<&DataArray<T>> for DataArray<T>
+where
+    T: DaftIntegerType,
+    <T as DaftNumericType>::Native: Ord,
+    <T as DaftNumericType>::Native: Hash,
+    <T as DaftNumericType>::Native: std::cmp::Eq,
+{
+    type Output = DaftResult<BooleanArray>;
+
+    fn is_in(&self, rhs: &DataArray<T>) -> Self::Output {
+        collect_to_hashset_and_check_membership!(self, rhs)
+    }
+}
+
+macro_rules! impl_is_in_floating_array {
+    ($arr:ident) => {
+        impl DaftIsIn<&$arr> for $arr {
+            type Output = DaftResult<BooleanArray>;
+
+            fn is_in(&self, rhs: &$arr) -> Self::Output {
+                // collect to vec because floats don't implement Hash
+                let vec = rhs.as_arrow().iter().collect::<Vec<_>>();
```

Comment:
You can use `BTreeSet` which has log(n) lookups and doesn't require Hash. However you may have to implement `Ord` on a `struct FloatWrapper` that wraps the f32/f64.

https://doc.rust-lang.org/std/collections/struct.BTreeSet.html

Reply:
just realised that we have a hashable_float_wrapper: https://github.com/Eventual-Inc/Daft/blob/main/src/daft-core/src/utils/hashable_float_wrapper.rs , I could implement `Eq` and `PartialEq` for this wrapper and be able to use HashSet, would this be alright?

Reply:
I would favor BTreeSet for this since floats don't hold the invariant that something that is equal will give the same hash value.